### ISSUE TITLE
Proposal: support @scope for @scope/eslint-plugin

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#9903]
**Tell us about your environment**

* **ESLint Version:** v4.16.0
* **Node Version:** v9.2.0
* **npm Version:** v5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

<details open>
<summary>Configuration</summary>

```js
module.exports = {
  plugins: [
    '@scope'
  ],
  rules: {
    '@scope/rule': 'error'
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I run ESLint.

**What did you expect to happen?**

ESLint loads `@scope/eslint-plugin`.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint failed to load `eslint-plugin-@scope`.

```
Oops! Something went wrong! :(

ESLint: 4.16.0.
ESLint couldn't find the plugin "eslint-plugin-@scope". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-@scope is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-@scope@latest --save-dev

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

----

ESLint supports `@scope` for `@scope/eslint-config` since #3123, but doesn't support `@scope` for `@scope/eslint-plugin`.
It looks inconsistent.

I propose the feature, ESLint resolves `plugins: ["@scope"]` to `@scope/eslint-plugin`.

**Comment by mysticatea** [comment]
I add this to `tsc-agenda` since this is a core feature.

**TSC Summary:** ESLint resolves `@scope` inconsistently:

- `extends: ["@scope"]` → `@scope/eslint-config`
- `plugins: ["@scope"]` → `eslint-plugin-@scope`

Because we have received the request "`@scope` → `@scope/eslint-config`" in #3123 but never received the request "`@scope` → `@scope/eslint-plugin`".

So I propose the feature that ESLint resolves "`plugins: ["@scope"]` to `@scope/eslint-plugin`" for consistency.

**TSC Question:** Should we accept this feature?

**Comment by not-an-aardvark** [comment]
What happens if `@foo/eslint-plugin` and `eslint-plugin-@foo` both exist?

```yml
plugins:
  - @foo
  - eslint-plugin-@foo
rules:
  @foo/bar: error
```

It seems like `@foo/eslint-plugin` and `eslint-plugin-@foo` would both be able to create rules under the `@foo` namespace.

**Comment by mysticatea** [comment]
I think that `eslint-plugin-@foo` is not valid name for npm packages:
https://github.com/npm/validate-npm-package-name/blob/master/index.js#L76

2018年2月2日(金) 1:00 Teddy Katz <notifications@github.com>:

**Comment by not-an-aardvark** [comment]
In [last week's TSC meeting](https://github.com/eslint/tsc-meetings/blob/b76a2f0ba1f8701602e7d5178f28b3a80bf98a27/notes/2018/2018-02-15.md#support-scope-for-scopeeslint-plugin), the TSC decided to accept this as a breaking change.
